symengine/functions.h

```diff
@@ -1037,6 +1037,55 @@ class Abs: public Function {
 
 //! Canonicalize Abs:
 RCP<const Basic> abs(const RCP<const Basic> &arg);
+
+class Max: public Function {
+
+private:
+    vec_basic arg_;
+public:
+    IMPLEMENT_TYPEID(MAX)
+    //! Max Constructor
+    Max(const vec_basic&& arg);
+    Max(const RCP<const Basic> &arg, ...);
+
+    virtual bool __eq__(const Basic &o) const;
+    virtual int compare(const Basic &o) const;
+    //! \return Size of the hash
+    virtual std::size_t __hash__() const;
+    //! \return `true` if canonical
+    bool is_canonical(const vec_basic &arg) const;
+    inline vec_basic get_args() const { return arg_; }
+};
+
+//! Canonicalize Max:
+RCP<const Basic> max(const vec_basic &arg);
+
+class Min: public Function {
+
+private:
+    vec_basic arg_;
+public:
+    IMPLEMENT_TYPEID(MAX)
+    //! Min Constructor
+    Min(const vec_basic&& arg);
+    Min(const RCP<const Basic> &arg, ...);
+
+    virtual bool __eq__(const Basic &o) const;
+    virtual int compare(const Basic &o) const;
+    //! \return Size of the hash
+    virtual std::size_t __hash__() const;
+    //! \return `true` if canonical
+    bool is_canonical(const vec_basic &arg) const;
+    inline vec_basic get_args() const { return arg_; }
+};
+
+//! Canonicalize Min:
+RCP<const Basic> min(const vec_basic &arg);
+
+//! Recursive function to fully unnest Min/Max
+template<class T>
+void get_unnested_args(const vec_basic& args, vec_basic& unnested_args);
```

Comment:
This cannot work in my opinion, because this function is defined in the `.cpp` file, but you use templates. So either you have to implement it here in the header file and then test it in tests, or only implement it in the .cpp file and remove it from the header file.